examples/morpheusvm/tests/integration/integration_test.go

```diff
@@ -212,7 +214,44 @@ var _ = ginkgo.BeforeSuite(func() {
 		toEngine := make(chan common.Message, 1)
 		db := memdb.New()
 
-		v, err := controller.New(vm.WithManualGossiper(), vm.WithManualBuilder())
+		txDBIndexer := indexer.NewTxDBIndexer(memdb.New())
+		indexerFactory := indexer.NewSubscriptionFactory(txDBIndexer)
+		indexerAPI := indexer.NewAPIFactory(txDBIndexer, "morpehusvm", "/indexer")
+
+		server, handler := rpc.NewWebSocketServer(
+			l,
+			trace.Noop,
+			lconsts.ActionRegistry,
+			lconsts.AuthRegistry,
+			10000000,
+		)
+
+		webSocketFactory := rpc.NewPubSubFactory(handler)
+		removeTxSubscription := rpc.SubscriptionFuncFactory[vm.TxRemovedEvent]{
+			AcceptF: func(event vm.TxRemovedEvent) error {
+				return server.RemoveTx(event.TxID, event.Err)
+			},
+		}
+
+		acceptBlockSubscription := &rpc.SubscriptionFuncFactory[*chain.StatelessBlock]{
+			AcceptF: func(event *chain.StatelessBlock) error {
+				return server.AcceptBlock(event)
+			},
+		}
+
+		v, err := controller.New(
+			vm.WithManualGossiper[*controller.Controller](),
+			vm.WithManualBuilder[*controller.Controller](),
+			vm.WithBlockSubscriptions[*controller.Controller](indexerFactory, acceptBlockSubscription),
+			vm.WithRemoveTxSubscriptions[*controller.Controller](removeTxSubscription),
+			vm.WithVMAPIs[*controller.Controller](
+				rpc.JSONRPCServerFactory{},
+				indexerAPI,
+				webSocketFactory,
+			),
+			vm.WithControllerAPIs[*controller.Controller](&lrpc.JSONRPCServerFactory{}),
```

Comment:
I think this is pretty weird. I don't think we should have a concept of "controller" apis. I think apis should just be registering subscriptions into the VM and the VM doesn't even know what a controller api is. I think we can refactor the existing controller apis which are used to get hypersdk metadata like balances into the VM apis since that's not something the controller defines anyways.